### ISSUE TITLE
Ray direction and world coordinate

### ISSUE THREAD

**Issue by charlietianx** [bmild/nerf#177]
Based on what I understand, we take world coordinate system's origin as (0, 0, 0). In the fern data, some rays' origin(which is the 3*1 translation in the transformation matrix), the z coordinate is negative, which means it's behind the scene. How is that possible?

Another question, let's say the c2w is 
_**x1 y1 z1 t1
x2 y2 z2 t2
x3 y3 z3 t3
0 0 0 1**_
Then for every camera, the 3 unit vector of their coordinate system's axis shoud be: 
_**x direction: c2w * (1, 0, 0, 1) = (x1 + t1, x2 + t2, x3 + t3)
y direction: (y1 + t1, y2 + t2, y3 + t3)
z direction: (z1 + t1, z2 + t2, z3 + t3)**_
Is this correct? I plot these camera coordinates in octave, it looks like this(the big one is world coordinate system), red, green, blue as x, y, z axis.
They just look weird, because the camera's coordinate system's y and z are not the same as world coordinate's. 
Shouldn't the blue dash arrow point to the reverse direction of the big blue arrow(because camera looks at the scene), and the green dash arrows point to the same direction as the big green arrow(because y axis points upwards)

![untitled](https://user-images.githubusercontent.com/12771564/218215486-7cc630c6-b129-4af7-b8b2-67715c5c420a.jpg)

**Comment by Dipankar1997161** [comment]
I think the c2w matrix that you mentioned, in that t1, specifically stands for translation along the x-axis and t2 for the y-axis. Therefore, the x2+t2 seems kinda inaccurate to me.
t2 should strictly be with the y-axis.

So, if you wish to calculate the Xc, Yc and Zc then it should be something like this:
Xc = x1 + y1 + z1 + t1
Yc = x2 + y2 + z2 +t2
Zc = x3 +y3 +z3 + t3

**Comment by charlietianx** [comment]
> I think the c2w matrix that you mentioned, in that t1, specifically stands for translation along the x-axis and t2 for the y-axis. Therefore, the x2+t2 seems kinda inaccurate to me. t2 should strictly be with the y-axis.
> 
> So, if you wish to calculate the Xc, Yc and Zc then it should be something like this: Xc = x1 + y1 + z1 + t1 Yc = x2 + y2 + z2 +t2 Zc = x3 +y3 +z3 + t3

Hi thank you for your comment. I used (x1 + t1, x2 + t2, x3 + t3) to calculate the unit 3 vectors of camera coordinate system. For example, the camera's 3 vectors should be (1, 0, 0), (0, 1, 0), (0, 0, 1). If we use the transformation matrix to calculate these 3 point in world coordinate, it will be (x1, x2, x3), (y1, y2, y3), (z1, z2, z3), if we don't consider the translation. 

My question is, what the world coordinate look like, compared to the camera coordinate? In nerf, isn't the world coordinate's (x, y, z)  directions are (right, up, forward) when we use the right hand coordiante system, sort of like this(the origin is the object we are looking at, and we are looking backwards compared to world's z axis):
![untitled](https://user-images.githubusercontent.com/12771564/218586151-ff153582-e646-495d-b9f8-3c5f3fc7c564.jpg)

**Comment by Dipankar1997161** [comment]
I think for your first question regarding the Negative Z direction. 

If I am not wrong the Z is negative for the camera viewing direction.
So camera's Negative Z defines that " the camera is pointing towards the Negative Z direction where the target is located" also known as camera's frame of referrence.

**Comment by charlietianx** [comment]
Hi, thank you for the reply. Now I understand the z axis. What about the y axis? I thought they would both point upwards. But  from some posts I found and also from my experiment in octave, they should seem like this(however I don't know how do we get to know, is it explained somewhere?):
![IMG_8905](https://user-images.githubusercontent.com/12771564/221364656-999b5509-27fa-495a-aa10-b13be44ecc0e.jpg)

**Comment by Dipankar1997161** [comment]
The y-axis in the camera coordinate usually defines the up vector or in simpler terms - top of the camera 

A negative of that value will only mean - the camera head ( or top ) is downwards. 

When you actually perform Nerf in this case, you will see the different render views when you change the y-axis from positive to negative.

**Comment by charlietianx** [comment]
> The y-axis in the camera coordinate usually defines the up vector or in simpler terms - top of the camera
> 
> A negative of that value will only mean - the camera head ( or top ) is downwards.
> 
> When you actually perform Nerf in this case, you will see the different render views when you change the y-axis from positive to negative.

Hi, the axis you are talking about is the camera space(where the origin is the point where the camera located). I want to know how the coordinate system of camera space look like with respect to the coordinate system of world space. This is the figure from the article I found, where x, y, z axises are in color R, G, B respectively.  I don't understand, why y axis point to world's -y direction:
![v2-5d890d500cacd9c1b3ddfc95327e5262_r](https://user-images.githubusercontent.com/12771564/221415468-1bc33a91-6a2f-42c0-a260-cb3a96a33303.jpg)

**Comment by charlietianx** [comment]
I got it now. With _poses_ being the real location and position of a camera, ofc they are coordinates regarding the world coordinate system(which has 3 axes x(1, 0, 0), y (0, 1, 0) and z(0, 0, 1))
